services/rds/rds.go

```diff
@@ -302,7 +305,21 @@ func (svc *Service) Discover(ctx context.Context, accessKey, secretKey string) (
 					return errors.WithStack(err)
 				}
 
-				out, err := rds.New(s).DescribeDBInstancesWithContext(ctx, new(rds.DescribeDBInstancesInput))
+				pageNum := 0
+				DBInstanceOutput := []*rds.DescribeDBInstancesOutput{}
+
+				err = rds.New(s).DescribeDBInstancesPagesWithContext(ctx, new(rds.DescribeDBInstancesInput),
+					func(page *rds.DescribeDBInstancesOutput, lastPage bool) bool {
+						pageNum++
+
+						DBInstanceOutput = append(DBInstanceOutput, page)
+						if lastPage {
+							return pageNum <= maxPageNum
+						}
+
+						return pageNum <= maxPageNum
```

Comment:
This condition is the same as in `if` above. This really looks like a bug. Either one of them is wrong, or `if` is no-op and should be removed.

Reply:
Updated